streamwebs_frontend/streamwebs/tests/views/test_water_quality_edit_view.py

```diff
@@ -0,0 +1,168 @@
+from django.test import Client, TestCase
+from django.core.urlresolvers import reverse
+from django.contrib.auth.models import User
+from streamwebs.models import Site
+
+
+class AddWaterQualityTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.user = User.objects.create_user(
+            'john', 'john@example.com', 'johnpassword'
+        )
+        self.client.login(username='john', password='johnpassword')
+
+    def test_view_with_bad_blank_data(self):
+        """
+        When the user tries to submit a bad (blank) form, the form errors
+        should be displayed
+        """
+        site = Site.test_objects.create_site('Site Name', 'Site Type', 'slug')
+        response = self.client.post(
+            reverse(
+                'streamwebs:water_quality_edit',
+                kwargs={'site_slug': site.site_slug}
+            ), {
+                'water_quality-TOTAL_FORMS': '4',
+                'water_quality-INITIAL_FORMS': '0',
+                'water_quality-MAX_NUM_FORMS': '4'
+                }
+        )
+        self.assertFormError(
+            # At least one field should return an error
+            response, 'wq_form',
+            'fish_present',
+            'This field is required.'
+        )
+        self.assertFalse(response.context['added'])
+
+    def test_view_with_good_data(self):
+        """
+        When the user submits a form with all required fields filled
+        appropriately, the user should see a success message
+        """
+        site = Site.test_objects.create_site('Site Name', 'Site Type', 'slug')
+        response = self.client.post(
+            reverse(
+                'streamwebs:water_quality_edit',
+                kwargs={'site_slug': site.site_slug}
+            ), {
+                'DEQ_dq_level': u'A',
+                'air_temp_unit': u'Fahrenheit',
+                'water_temp_unit': u'Fahrenheit',
+                'csrfmiddlewaretoken': u'vyAy6MjcVOUXCJKvBIr1VmQb8el2ppoL',
+                'date': u'2016-08-22',
+                'dead_fish': 2,
+                'fish_present': u'True',
+                'initial-date': u'2016-08-22',
+                'latitude': 125.16,
```

Comment:
After doing some reading, I think I MIGHT understand how this works.  Please correct me if I'm wrong.

In geodjango, Point takes (Long, Lat) (for reasons unknown) and internally makes them accessible as (x, y).  That means whichever is passed in first becomes x, and whichever second becomes y.

If we have been doing that all along, then all of our values in the database would be reversed and lat=x, long=y.

Is that pretty much the gist of it?  I see a lot of examples where lat=y, long=x, but they are passing in the values as (long, lat) and we aren't.